sunspot/lib/sunspot/indexer.rb

```diff
@@ -54,9 +54,19 @@ def remove(*models)
     # Remove the model from the Solr index by specifying the class and ID
     #
     def remove_by_id(class_name, *ids)
+      id_prefix = nil
```

Comment:
I think the value of this var should better be assigned below, with a return value of `if ` conditional

Reply:
@serggl this one is fixed too